docarray/index/backends/in_memory.py

```diff
@@ -0,0 +1,304 @@
+        """
+        indices = []
+        for i, doc in enumerate(self._docs):
+            if doc.id in doc_ids:
+                indices.append(i)
+        return self._docs[indices]
+
+    def execute_query(self, query: List[Tuple[str, Dict]], *args, **kwargs) -> Any:
+        """
+        Execute a query on the InMemoryDocIndex.
+
+        Can take two kinds of inputs:
+
+        1. A native query of the underlying database. This is meant as a passthrough so that you
+        can enjoy any functionality that is not available through the Document index API.
+        2. The output of this Document index' `QueryBuilder.build()` method.
+
+        :param query: the query to execute
+        :param args: positional arguments to pass to the query
+        :param kwargs: keyword arguments to pass to the query
+        :return: the result of the query
+        """
+        if args or kwargs:
+            raise ValueError(
+                f'args and kwargs not supported for `execute_query` on {type(self)}'
+            )
+
+        ann_docs = DocList.__class_getitem__(cast(Type[BaseDoc], self._schema))([])
+        filter_conditions = []
+        doc_to_score: Dict[BaseDoc, Any] = {}
+        for op, op_kwargs in query:
+            if op == 'find':
+                docs, scores = self.find(**op_kwargs)
+                ann_docs.extend(docs)
+                doc_to_score.update(zip(docs.__getattribute__('id'), scores))
+            elif op == 'filter':
+                filter_conditions.append(op_kwargs['filter_query'])
+
+        self._logger.debug(f'Executing query {query}')
+        docs_filtered = ann_docs
+        for cond in filter_conditions:
+            docs_cls = DocList.__class_getitem__(cast(Type[BaseDoc], self._schema))
+            docs_filtered = docs_cls(filter_docs(docs_filtered, cond))
+
+        self._logger.debug(f'{len(docs_filtered)} results found')
+        docs_and_scores = zip(
+            docs_filtered, (doc_to_score[doc.id] for doc in docs_filtered)
+        )
+        docs_sorted = sorted(docs_and_scores, key=lambda x: x[1])
+        out_docs, out_scores = zip(*docs_sorted)
+        return FindResult(documents=out_docs, scores=out_scores)
+
+    def find(
+        self,
+        query: Union[AnyTensor, BaseDoc],
+        search_field: str = '',
+        limit: int = 10,
+        **kwargs,
+    ) -> FindResult:
+        """Find documents in the index using nearest neighbor search.
+
+        :param query: query vector for KNN/ANN search.
+            Can be either a tensor-like (np.array, torch.Tensor, etc.)
+            with a single axis, or a Document
+        :param search_field: name of the field to search on.
+            Documents in the index are retrieved based on this similarity
+            of this field to the query.
+        :param limit: maximum number of documents to return
```

Comment:
```suggestion
        :param limit: maximum number of Documents to return
```
Ensure consistent capitalization throughout please :)